commit 61ed46048d3ae7aa182102d192bab3e49339a010
Author: Volker Krause <vkrause@kde.org>
Date:   Sun Jul 30 15:47:29 2017 +0200

    Allow to build KConfig without Qt5Gui
    
    This is particularly useful for cross-compilation, where we only need the
    kconfig_compiler on the host system.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5e50ba5..458630e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,8 @@ set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
 
 set(REQUIRED_QT_VERSION 5.6.0)
 
-find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Gui Xml)
+find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Xml)
+find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG OPTIONAL_COMPONENTS Gui)
 include(KDEInstallDirs)
 include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
 include(KDECMakeSettings)
diff --git a/autotests/CMakeLists.txt b/autotests/CMakeLists.txt
index c1ae1de..7d6698f 100644
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -52,6 +52,7 @@ ecm_add_tests(
 
 target_include_directories(test_kconf_update PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/../src/kconf_update)
 
+if(TARGET Qt5::Gui)
 ecm_add_tests(
    kconfigguitest.cpp
    kconfigloadertest.cpp
@@ -62,4 +63,4 @@ ecm_add_tests(
 )
 
 add_subdirectory(kconfig_compiler)
-
+endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 998a1d5..d4ca886 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,7 @@
 add_subdirectory(core)
-add_subdirectory(gui)
+if(TARGET Qt5::Gui)
+    add_subdirectory(gui)
+endif()
 add_subdirectory(kconfig_compiler)
 add_subdirectory(kconf_update)
 add_subdirectory(kreadconfig)
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 9b7491c..46fabd5 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -1,6 +1,3 @@
-
-find_package(Qt5Core ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
-
 set(libkconfigcore_SRCS
    kconfig.cpp
    kconfigbase.cpp
diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index 9659326..08786d3 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -1,7 +1,3 @@
-
-find_package(Qt5Gui ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
-find_package(Qt5Xml ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
-
 set(libkconfiggui_SRCS
    kconfiggui.cpp
    kconfiggroupgui.cpp
diff --git a/src/kconf_update/CMakeLists.txt b/src/kconf_update/CMakeLists.txt
index b1342a8..db11511 100644
--- a/src/kconf_update/CMakeLists.txt
+++ b/src/kconf_update/CMakeLists.txt
@@ -1,5 +1,3 @@
-find_package(Qt5Core ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
-
 remove_definitions(-DQT_NO_CAST_FROM_ASCII)
 
 ########### next target ###############
diff --git a/src/kconfig_compiler/CMakeLists.txt b/src/kconfig_compiler/CMakeLists.txt
index 004a649..dc0a08d 100644
--- a/src/kconfig_compiler/CMakeLists.txt
+++ b/src/kconfig_compiler/CMakeLists.txt
@@ -16,8 +16,6 @@ else()
     add_executable(KF5::kconfig_compiler ALIAS kconfig_compiler)
 endif()
 
-find_package(Qt5Xml ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
-
 target_link_libraries(kconfig_compiler Qt5::Xml)
 
 ecm_mark_nongui_executable(kconfig_compiler)
